diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS c-blosc2-2.1.1.orig/blosc/blosc2.c c-blosc2-2.1.1/blosc/blosc2.c
--- c-blosc2-2.1.1.orig/blosc/blosc2.c	2022-05-07 12:51:50.000000000 +0200
+++ c-blosc2-2.1.1/blosc/blosc2.c	2022-05-15 21:48:54.381758721 +0200
@@ -435,15 +435,19 @@
 static int zlib_wrap_compress(const char* input, size_t input_length,
                               char* output, size_t maxout, int clevel) {
   int status;
-  uLongf cl = (uLongf)maxout;
 #ifdef ZLIB_COMPAT
+  uLongf cl = (uLongf)maxout;
   status = compress2(
+      (Bytef*)output, &cl, (Bytef*)input, (uLong)input_length, clevel);
 #elif defined(HAVE_ZLIB_NG)
+  size_t cl = maxout;
   status = zng_compress2(
+      (uint8_t*)output, &cl, (uint8_t*)input, input_length, clevel);
 #else
+  uLongf cl = (uLongf)maxout;
   status = compress2(
-#endif
       (Bytef*)output, &cl, (Bytef*)input, (uLong)input_length, clevel);
+#endif
   if (status != Z_OK) {
     return 0;
   }
@@ -453,15 +457,19 @@
 static int zlib_wrap_decompress(const char* input, size_t compressed_length,
                                 char* output, size_t maxout) {
   int status;
-  uLongf ul = (uLongf)maxout;
 #ifdef ZLIB_COMPAT
+  uLongf ul = (uLongf)maxout;
   status = uncompress(
+      (Bytef*)output, &ul, (Bytef*)input, (uLong)compressed_length);
 #elif defined(HAVE_ZLIB_NG)
+  size_t ul = maxout;
   status = zng_uncompress(
+      (uint8_t*)output, &ul, (uint8_t*)input, compressed_length);
 #else
+  uLongf ul = (uLongf)maxout;
   status = uncompress(
-#endif
       (Bytef*)output, &ul, (Bytef*)input, (uLong)compressed_length);
+#endif
   if (status != Z_OK) {
     return 0;
   }
diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS c-blosc2-2.1.1.orig/compat/filegen.c c-blosc2-2.1.1/compat/filegen.c
--- c-blosc2-2.1.1.orig/compat/filegen.c	2022-05-07 12:51:50.000000000 +0200
+++ c-blosc2-2.1.1/compat/filegen.c	2022-05-15 21:29:11.284903070 +0200
@@ -33,6 +33,7 @@
 
 
 int main(int argc, char *argv[]) {
+  setvbuf(stdout, NULL, _IONBF, 0);
   BLOSC_UNUSED_PARAM(argc);
   static int32_t data[SIZE];
   static int32_t data_out[SIZE];
diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS c-blosc2-2.1.1.orig/tests/CMakeLists.txt c-blosc2-2.1.1/tests/CMakeLists.txt
--- c-blosc2-2.1.1.orig/tests/CMakeLists.txt	2022-05-07 12:51:50.000000000 +0200
+++ c-blosc2-2.1.1/tests/CMakeLists.txt	2022-05-11 19:10:40.206391532 +0200
@@ -16,6 +16,7 @@
 
     # Some tests will be enabled only for Unix
     if(WIN32)
+        continue()
         if(target STREQUAL test_nolock OR
             target STREQUAL test_noinit OR
             target STREQUAL test_compressor OR
