diff --git a/blosc/bitshuffle-altivec.h b/blosc/bitshuffle-altivec.h
index a7f1c081..7416c7a8 100644
--- a/blosc/bitshuffle-altivec.h
+++ b/blosc/bitshuffle-altivec.h
@@ -13,7 +13,7 @@
 #ifndef BITSHUFFLE_ALTIVEC_H
 #define BITSHUFFLE_ALTIVEC_H
 
-#include "blosc2-common.h"
+#include "blosc2/blosc2-common.h"
 
 #ifdef __cplusplus
 extern "C" {
diff --git a/blosc/shuffle-altivec.h b/blosc/shuffle-altivec.h
index 557eb966..3825557e 100644
--- a/blosc/shuffle-altivec.h
+++ b/blosc/shuffle-altivec.h
@@ -13,7 +13,7 @@
 #ifndef SHUFFLE_ALTIVEC_H
 #define SHUFFLE_ALTIVEC_H
 
-#include "blosc2-common.h"
+#include "blosc2/blosc2-common.h"
 
 #ifdef __cplusplus
 extern "C" {
