diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS -x autom4te.cache c-blosc2-2.9.2.orig/plugins/codecs/ndlz/CMakeLists.txt c-blosc2-2.9.2/plugins/codecs/ndlz/CMakeLists.txt
--- c-blosc2-2.9.2.orig/plugins/codecs/ndlz/CMakeLists.txt	2023-05-25 17:41:09.086025927 +0200
+++ c-blosc2-2.9.2/plugins/codecs/ndlz/CMakeLists.txt	2023-05-25 17:45:27.761312368 +0200
@@ -23,7 +23,7 @@
             TARGET test_ndlz
             APPEND PROPERTY COMPILE_DEFINITIONS BLOSC_TESTING)
 
-    target_link_libraries(test_ndlz blosc_testing)
+    target_link_libraries(test_ndlz blosc_testing ${CMAKE_DL_LIBS})
 
     # tests
         add_test(NAME test_plugin_test_ndlz
